taqueria-plugin-smartpy/compile.ts

```diff
@@ -1,66 +1,185 @@
-import { execCmd, getContracts, sendAsyncJsonRes, sendErr } from '@taqueria/node-sdk';
-import { RequestArgs, TaqError } from '@taqueria/node-sdk/types';
-import { readFile } from 'fs/promises';
-import { join } from 'path';
-
-interface Opts extends RequestArgs.ProxyRequestArgs {
-	sourceFile?: string;
-}
-
-const getArtifacts = (sourceAbspath: string) => {
-	return readFile(sourceAbspath, { encoding: 'utf-8' })
-		.then((source: string) => {
-			const pattern = new RegExp(/add_compilation_target\s*\(\s*(['"])([^'"]+)\1/, 'mg');
-			const results = source.matchAll(pattern);
-			return results
-				? Array.from(results).map(match => match[2])
-				: ['--'];
-		});
+import { execCmd, getArch, sendAsyncErr, sendJsonRes, sendWarn } from '@taqueria/node-sdk';
+import { copyFile, readFile } from 'fs/promises';
+import { basename, extname, join } from 'path';
+import { CompileOpts as Opts, emitExternalError, getCompilationTargetsDirName, getInputFilename } from './common';
+
+type TableRow = { contract: string; artifact: string };
+
+const COMPILE_ERR_MSG: string = 'Not compiled';
+
+const addPyExtensionIfMissing = (sourceFile: string): string => {
+	return /\.py$/.test(sourceFile) ? sourceFile : `${sourceFile}.py`;
+};
+
+const getOutputContractFilename = (parsedArgs: Opts, sourceFile: string): string => {
+	const outputFile = basename(sourceFile, extname(sourceFile));
```

Comment:
I think the function `basename` will just do the same if you don't pass the second argument to it? And will be simpler.

Reply:
It won't do the same. The second argument is needed

Reply:
```ts
            /**
             * Return the last portion of a path. Similar to the Unix basename command.
             * Often used to extract the file name from a fully qualified path.
             *
             * @param path the path to evaluate.
             * @param suffix optionally, an extension to remove from the result.
             * @throws {TypeError} if `path` is not a string or if `ext` is given and is not a string.
             */
            basename(path: string, suffix?: string): string;
```

Reply:
Yep there `@param suffix optionally, an extension to remove from the result.`

path1/path2/file.mligo becomes file.mligo but with that 2nd argument, it becomes file, which is what I want